processor/metricstransformprocessor/metrics_transform_processor_helper.go

```diff
@@ -0,0 +1,375 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metricstransformprocessor
+
+import (
+	"fmt"
+	"math"
+
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	"github.com/golang/protobuf/ptypes/timestamp"
+)
+
+// update updates the metric content based on operations indicated in transform.
+func (mtp *metricsTransformProcessor) update(metric *metricspb.Metric, transform Transform) {
+	// metric name update
+	if transform.NewName != "" {
+		metric.MetricDescriptor.Name = transform.NewName
+	}
+
+	for _, op := range transform.Operations {
+		switch op.Action {
+		case UpdateLabel:
+			mtp.updateLabelOp(metric, op)
+		case AggregateLabels, AggregateLabelValues:
+			mtp.aggregateOp(metric, op)
+		}
+	}
+}
+
+// updateLabelOp updates labels and label values in metric based on given operation
+func (mtp *metricsTransformProcessor) updateLabelOp(metric *metricspb.Metric, op Operation) {
+	for idx, label := range metric.MetricDescriptor.LabelKeys {
+		if label.Key != op.Label {
+			continue
+		}
+		// label key update
+		if op.NewLabel != "" {
+			label.Key = op.NewLabel
+		}
+		// label value update
+		labelValuesMapping := mtp.createLabelValueMapping(op.ValueActions)
```

Comment:
I am not sure about what you mean by "on startup", this mapping creation step is only needed when I have a updateLabel operation and on the corresponding valueActions. I don't see how this map creation step would be unnecessarily repeated other than if the user wants to rename the same value twice?

Reply:
This might only be generated once per call to ConsumeMetrics, but ConsumeMetrics will be called many times during the lifetime of the collector

Reply:
Based on my understanding, ConsumeMetrics is called once per processor right? And I think the label values mapping is created here because this is only needed as part of the operations needed for label value rename. Otherwise, where do you think would be a good place to build this mapping?

Reply:
ConsumeMetrics will get called once for each processor for each batch of metrics that gets processed.

For example, if you're using a pipline with the host metrics receiver with a 1m scrape interval and the metrics trasnform processor, then ConsumeMetrics will get called every minute with the same metric metadata, but different values.

If this only ran once, then how you have it would probably be best (keep the object close to where it's used), but instead I think you should create this structure on initialization, probably in the `Start` function, and store it in the scraper,